This is for code review only - NOT FOR MERGE (Related to RavenDB-12503, RavenDB-12206 and RavenDB-12187)

myarichuk opened the pull request:
1) Make sure that filters & select in edge will work when there is string array as edge property
2) Add unit test for RavenDB-12187 - skipped for now, will be "unskipped" as the issue is fixed
3) Unskip & fix several unit tests that were skipped before (they were skipped because of unimplemented issues)

myarichuk committed:
This is for code review only - NOT FOR MERGE:
1) Make sure that filters & select in edge will work when there is string array as edge property
2) Add unit test for RavenDB-12187 - skipped for now, will be "unskipped" as the issue is fixed
3) Unskip & fix several unit tests that were skipped before (they were skipped because of unimplemented issues)
(RavenDB-12503 and RavenDB-12206)

ayende commented on src/Raven.Server/Documents/Queries/Graph/SingleEdgeMatcher.cs:
```diff
@@ -44,27 +45,59 @@ public void SingleMatch(Match left, string alias)
+                                    AddEdgeAfterFiltering(left, json, projectFieldValue);
+                                    break;
+                                case LazyStringValue lazyString:
+                                    var jsonStringForWhere = new DynamicJsonValue
```
Don't we need to check that `projectFieldValue` is null here?

myarichuk replied:
Probably (will need to test if it can become null and how), and also we need to test for arrays of primitives that are not strings.
I did this PR because I am not 100% sure that the code so far is good enough - and I mean the code around IsMatchedBy() that I added

myarichuk commented:
added another commit, better handling of use-cases for RavenDB-12206
(planning to submit separate PR for RavenDB-12206, as soon as I synchronize with Tal's code changes)

myarichuk committed:
In graph queries
1) make sure to properly handle non-string arrays as edge properties.
2) some improvements in handling string arrays as edges

(RavenDB-12206)

myarichuk commented:
rewrote the commit, make sure all tests are passing

myarichuk commented:
will submit stuff as per-issue PRs
